dev/integration_tests/flutter_gallery/lib/demo/material/text_form_field_demo.dart

```diff
@@ -143,10 +143,9 @@ class TextFormFieldDemoState extends State<TextFormFieldDemo> {
     return null;
   }
 
-  Future<bool> _warnUserAboutInvalidData() async {
-    final FormState? form = _formKey.currentState;
-    if (form == null || !_formWasEdited || form.validate()) {
-      return true;
+  Future<void> _onPopped(bool success) async {
+    if (success) {
```

Comment:
In all of these examples that I've done for PopScope, I've always just returned when `success` is true.  However, I imagine some people might want to use this to respond to a back gesture even if it was successful (maybe they want to save something?).  It follows the previous API before this PR, WillPopScope.onWillPop.

Can you explain more about your points on sync/async?  The code that calls onPopped is in routes.dart for the record.  It can be triggered asynchronously after an `await` via Navigator.maybePop.

Reply:
nvm, I think you are right, it is probably better to keep this.

for how to split out the callback, I don't have a preference. you can even remove the success boolean here since it is the developer who provides the the popEnabled, so they should have known the answer already, and they need a set state to update the boolean, they may as well swap the callback at the same time

Reply:
Sounds good, I'll keep it as-is for now.

It's theoretically possible for `success` to disagree with `popEnabled`, for example, if there were two PopScopes in the same route with different `popEnabled`s.  Plus I think the existence of the `success` boolean helps communicate to users that the pop already happened and it's not possible to stop it.